source/plugins/ipv6.rst

```diff
@@ -199,3 +202,117 @@ each of the VMs. Use DUID-LL to set up dhclient.
+
+
+Adding a Public IPv6 Range
+##########################
+
+The administrator can use both UI and API to add a public IPv6 range. UI is the preferable option.
+Option to add a new public Ipv6 range in the UI can be found in Infrastructure -> Zones -> Zone details -> Physical Network tab -> Physical network details -> Traffic Types tab -> Public -> *Add IP range*.
+In the Add IP range form, IPv6 can be selected as the IP Range Type. IPv6 Gateway and CIDR must be provided and optionally a VLAN/VNI can be provided.
+
+Alternatively, ``createVlanIpRange`` API can be used to add a new public IPv6 range.
+
+|add-public-ipv6-range-form.png|
+
+
+
+   .. note::
+      As SLAAC based public IPv6 address will be assigned to the networks therefore public IPv6 range must be added without specifying start and end IP addresses.
+
+
+Adding Guest IPv6 Prefix
+########################
+
+Again, both UI and API to add a guest IPv6 prefix. UI is the preferable option.
+Option to add a new public Ipv6 range in the UI can be found in Infrastructure -> Zones -> Zone details -> Physical Network tab -> Physical network details -> Traffic Types tab -> Guest -> *Add IPv6 prefix*.
+In the Add IPv6 prefix form, an IPv6 prefix with CIDR size lesser than 64 must be provided.
+
+Alternatively, ``createGuestNetworkIpv6Prefix`` API can be used to add a new guest IPv6 prefix.
+
+|add-guest-ipv6-prefix-form.png|
+
+
+Adding Network Offering with IPv6 Support
+#########################################
+
+To create an IPv6 suported network offering, global configuration - ``network.offering.ipv6.enabled`` must be set to **true**.
+
+With 4.17.0, a new paramter - ``internetprotocol`` has been added to the ``createNetworkOffering`` API which can be used to create a network offering with IPv6 support by using the value dualstack.
+Corresponding option has also been provided in the UI form creating network offering:
+
+|add-ipv6-network-offering-form.png|
+
+
+Adding Upstream Route
+#####################
+
+Currently, CloudStack supports IPv6 isolated networks and VPC tiers only with **static** routes and therefore the administrator needs to add upstream IPv6 routes once a network is successfully deployed.
+To facilitate the automation, *CloudStack Event Notification* can be used. CloudStack will generate appropriate events on network creation or deletion and while assigning or releasing a public IPv6 address for a network. Based on the events the corresponding network can be queried for the IPv6 routes that it needs configured in upstream network.
+Upstream IPv6 routes required by an IPv6 supported isolated network or VPC tier are also shown in the UI in the network details.
+
+|network-details-upstream-ipv6-routes.png|
+
+
+IPv6 Firewall
```

Comment:
Do we have firewall enabled/disabled option in network offerings ?
what's the default policy ?

Reply:
@weizhouapache no separate option right now. It will depend on Firewall service enabled for the network.
Also, default policy for egress requests will be governed by `egressdefaultpolicy` of the offering. Ingress it is deny